src/include/duckdb/common/file_opener.hpp

```diff
@@ -16,16 +16,22 @@ namespace duckdb {
 class ClientContext;
 class Value;
 
+struct FileOpenerInfo {
+	string file_path;
+};
+
 //! Abstract type that provide client-specific context to FileSystem.
 class FileOpener {
 public:
+	FileOpener() {
+	}
 	virtual ~FileOpener() {};
 
-	virtual bool TryGetCurrentSetting(const string &key, Value &result) = 0;
+	virtual bool TryGetCurrentSetting(const string &key, Value &result, FileOpenerInfo &info) = 0;
```

Comment:
Can make new argument an optional_ptr and default it to a nullptr for most subclasses.

Reply:
Looking at usage again. It might be cleaner to have two virtual `TryGetCurrentSetting` methods one with the `FileOpenerInfo` and one without. This way users can only provide FileOpenerInfo when it makes sense, and avoids potentially confusing nullptrs

Reply:
Okay makes sense!

Reply:
Can default this method to disregard `FileOpenerInfo` to make life easier on library writer (though I am personally inclined towards forcing libraries to be explicit).